src/sgame/sg_team.cpp

```diff
@@ -569,23 +564,44 @@ void CheckTeamStatus()
 	{
 		level.lastTeamImbalancedTime = level.time;
 
-		if ( level.team[ TEAM_ALIENS ].numSpawns > 0 &&
-		     level.team[ TEAM_HUMANS ].numClients - level.team[ TEAM_ALIENS ].numClients > 2 )
+		int minPlayers = INT_MAX;
+		int maxPlayers = INT_MIN;
+		int lowerTeam = TEAM_NONE;
+		int upperTeam = TEAM_NONE;
+		int numTeamWithSpawns = 0;
+		for ( int team = TEAM_NONE; team < NUM_TEAMS; ++team )
 		{
-			trap_SendServerCommand( -1, "print_tr \"" N_("Teams are imbalanced. "
-			                        "Humans have more players.") "\"" );
-			level.numTeamImbalanceWarnings++;
+			int spawns  = level.team[ team ].numSpawns;
+			int clients = level.team[ team ].numClients;
+			if ( spawns <= 0 )
+			{
+				continue;
+			}
+			++numTeamWithSpawns;
+			if ( clients >= maxPlayers )
+			{
+				maxPlayers = clients;
+				upperTeam  = team;
+			}
+			if ( clients <= minPlayers )
+			{
+				minPlayers = clients;
+				lowerTeam  = team;
+			}
 		}
-		else if ( level.team[ TEAM_HUMANS ].numSpawns > 0 &&
-		          level.team[ TEAM_ALIENS ].numClients - level.team[ TEAM_HUMANS ].numClients > 2 )
+		ASSERT( numTeamWithSpawns < 2 || minPlayers != INT_MAX );
+		ASSERT( numTeamWithSpawns < 2 || maxPlayers != INT_MIN );
+		if ( numTeamWithSpawns >= 2 && maxPlayers - minPlayers > 2 )
 		{
-			trap_SendServerCommand( -1, "print_tr \"" N_("Teams are imbalanced. "
-			                        "Aliens have more players.") "\"" );
+			//TODO: build the "foo have less, bar have more" string
+			trap_SendServerCommand( -1, g_teamBalanceStrong[upperTeam] );
+			trap_SendServerCommand( -1, g_teamBalanceWeak[lowerTeam] );
 			level.numTeamImbalanceWarnings++;
 		}
-		else
-		{
-			level.numTeamImbalanceWarnings = 0;
-		}
 	}
 }
+
+int G_NumberPlayableTeams()
+{
+	return TEAM_HUMANS - TEAM_ALIENS + 1;
```

Comment:
humans are 2, aliens are 1, current code thus returns 2 -1 + 1 = 2

Reply:
and NUM_TEAMS is 3, thus 3-1 = 2. I'm not saying it's wrong right now, but if someone were to write this
```diff
 enum team_t
 {
   TEAM_ALL = -1,
   TEAM_NONE,
   TEAM_ALIENS,
   TEAM_HUMANS,
+  TEAM_ZOMBIES,
 
   NUM_TEAMS
 };
```
it would break :-/

Reply:
right, shuold use `NUM_TEAMS - 1` even if that would also break under some conditions, but without making team_t (or another struct for less problems) a structure that describe teams, and have a list of those somewhere, there's no real solution.